staging/src/k8s.io/mount-utils/mount.go

```diff
@@ -147,6 +147,23 @@ func NewMountError(mountErrorValue MountErrorType, format string, args ...interf
 type SafeFormatAndMount struct {
 	Interface
 	Exec utilexec.Interface
+
+	formatSem     chan any
+	formatTimeout time.Duration
+}
+
+// SetMaxConcurrentFormat sets the maximum number of concurrent format
+// operations executed by the mounter. The timeout controls the maximum
+// duration of a format operation before it is treated as complete. A max
```

Comment:
is a timed out format operation considered complete and *successful* or complete and *failed*? is a timed out format operation interrupted or allowed to continue in the background? if it is interrupted, does that have the potential to corrupt anything? is there a way to set concurrent operations without setting a timeout?

Reply:
Good questions. PTAL and let me know if it's clarified.

> Is there a way to set concurrent operations without setting a timeout?

I implemented it without a timeout originally, but requiring a timeout seems like the safest strategy to avoid gridlock (e.g. from a stuck format operation). Do you think we should make it optional?